src/writer/creator.cpp

```diff
@@ -220,14 +220,36 @@ namespace zim
 
       TPROGRESS();
 
+
+      for(auto& handler:data->m_handlers) {
+        // This silently create all the needed dirents
+        auto dirent = handler->getUniqueDirent();
+      }
+
+      // Now we have all the dirents (but not the data), we must correctly set/fix the dirents
+      // before we ask data to the handlers
+      TINFO("ResolveRedirectIndexes");
+      data->resolveRedirectIndexes();
+
+      TINFO("Set entry indexes");
+      data->setEntryIndexes();
+
+      TINFO("Resolve mimetype");
+      data->resolveMimeTypes();
+
+      TINFO("create title index");
+      data->createTitleIndex();
+
+      // We can now stop the dirents, and get their content
       for(auto& handler:data->m_handlers) {
         handler->stop();
-        auto dirent = handler->getDirent();
+        auto dirent = handler->getUniqueDirent();
         auto provider = handler->getContentProvider();
+        auto providerSize = provider->getSize();
```

Comment:
`providerSize` is unused. Does `provider->getSize()` have a significant side-effect?

Reply:
Code moved in #493